hasktorch/src/Torch/Functional.hs

```diff
@@ -566,7 +566,15 @@ maxPool2d
     -> Tensor -- ^ output
 maxPool2d kernelSize stride padding dilation ceilMode self =
     unsafePerformIO $ (cast6 ATen.max_pool2d_tllllb)
-        self kernelSize stride padding dilation ceilMode
+        self 
+        (asList kernelSize)
+        (asList stride)
+        (asList padding)
+        (asList dilation)
+        ceilMode
+        where
+            asList :: (Int, Int) -> [Int]
+            asList (a0, a1) = [a0, a1]
```

Comment:
I've run into this elsewhere, would be preferable if we could add tuples to the casting mechanism although in this context it's a pretty minor thing (any thoughts @junjihashimoto ?)

Anyway out of scope to have to resolve in this PR.